tests/unit/utils/request.js

```diff
@@ -0,0 +1,414 @@
+57V+85fUYdOFTfAp2f3ti+MkXTpBeeawDkLYqMYnwI1KJB1SgXUj2mSjxBObF79Y
+dCrSF0+uFtOOGy/q8+kgCv3P72UD0iHaQlWJCmRszxAbLwUTxDcb3KrROt6J+uo2
+c2IXTZZKtl8ws8cOMhy8FhGZ08R1waTv/AM9RW8Af9wKmE4gWZKwMEkIiwgXbIpP
+XDzYCygidKezKa38lNvxr2g=
+-----END CERTIFICATE-----`;
+
+const testKey = `-----BEGIN PRIVATE KEY-----
+MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCtG3i4ruQ4PwYp
+Qo3cZisOSuq0THWSjZgMzdUagWyO6x/KEVFrkqPgP+H+St9Wrvcnb8tUBt37b5yN
+hTG51C+2y32TF6hbsuWVLUppHy0hliIDO0iRDlZYpzNP5Fz0JbWTRwIzwwlMkmQx
+kef7ZYnd2gcDZJxSbfWqpJyFHwdCzDhKqGA2KToEAZ2XyC4wuAveyAhBFgPJ6kI8
+u7aI4wnTbGlL5H96YfBe2pm1I6fU+alXg4avu2jvQq3P9jH2yyBWmKMdSeVRxT7e
+h5e7lrs1Nlm/K6bKGdSNVoQof4HffeTS6TQM9EXaRXAOdZpIhve8MGQl6nHleGqt
+IP6P06/zAgMBAAECggEAYJ8IzuyvbcWfxr/jNrXAKoYeYuyaY2atC9iHrfe3hD4w
+zDTGYWjEF5MQrUNVTajzQwvfTdNXa9RsaYGCs9p/l2QVf1ihHC3Kf218Lxi7tBd8
+fJlGc9Cw2WLZ/SnVyGCT7NShogqm0hqT2ic6eNPAFBhx+a7aohfGG60twBAFmX5A
+MbUSo9/+o8Wfd8RYyohLKOvRS044gaIpYchu0VV/B0hTZIIG7Kf1RAb32NtDIi/9
+d5oqjTSPYHLEjuAaGXU0HM1CwhdKwvuIwMi3PYGSETPaWXxrTD/oHYhejbJHb8JP
+SSXUf9ZAVlQi7ugSaNPzRSYxv8adhN9F6XAGgdCgwQKBgQDGUiQWKwntCGHw1kdc
+xQwUyRlYUXARQ8aSDmDjn2tj+5nxE2sSA+e08fYKxXnpgz9Z5kLdoOFmIM/p7VWg
+iObuO4tKyOTVjeFKmgGAB6kuXRTo/78gJv3T3jMTAJSsejQAhUEGVi8SqGzi7/+b
+5l5yF2gBc9SaszxJb7ey/b25+QKBgQDfdBIoVSyp+o7iU1mt5559DYP0KYrEXSNq
+SCbrelTLH/c7+Q8rIKhLvuJKsHz0OXauOyTR4HkCdqVXn4W3LnaP+0xHAZqUPgJ/
+MmbaC7WtohcAgz8koJ+gEZaFvqDPnWARvZEg4Zvoj2WLnTG1XshV/bxT/DzFOmAc
+I94j5BzUSwKBgG3xo6fWfE4303QclrtLUgND6RUZMLOhizf1WKlX+8UV+qW3SihW
+meGqMeiOyaM266v/Bxqu0cY4cosQ0+OHgd6YjAQNky8A1ODyt9ouZRAa8jipb193
+vkfyawYh0Eo+BQ400XOd28LQNG0q992JvNorN7F2cWrB6q4bjQ9htAihAoGAOG6Q
+01zHXdooUaIpKNEw5nYOWBBRukunUQUNxzRqy62z+5JnsUWbGx4G+kPeGrOUdGX9
+Y3+kL+oU0a84zs1OJgiZ9+jAyCVs7gCllvWUGVixJHEA9lgWWA95CyfcZvJgu7o7
+N6mifTYRuBOn5R0dzRG6iR9PnaOjeBfa1weZ/EcCgYAxW9DQxAKCkmLQT/Z1IKZm
+BIw3g1Mc7lMQJUl0w80iXPjf77PtDsR2X+y49o5/DNbU4BV4AJZfkQEe5TjXo01U
+Bk6Qaz6E4mH3iBT2upTuh0+AkKgyz8lh93idIWGZtaip1yRBZYU0I6WfFVfGxGNd
+qpUgAEJjy9v59F76feVtJA==
+-----END PRIVATE KEY-----`;
+
+const postJson = { key: 'value' };
+const postJsonStringified = JSON.stringify(postJson);
+const postData = 'postrequest';
+
+function respondWithError(req, res, code, message) {
+    /* eslint-disable no-param-reassign */
+    res.statusCode = code;
+    res.statusMessage = message;
+    res.end();
+    /* eslint-enable no-param-reassign */
+}
+
+function respondWithValue(req, res, value) {
+    res.write(value);
+    res.end();
+}
+
+function handlePostRequest(req, res, expected) {
+    let rawData = '';
+    req.on('data', data => {
+        rawData += data;
+    });
+    req.on('end', () => {
+        if (rawData !== expected) {
+            return respondWithError(req, res, 400, 'incorrect body value');
+        }
+        res.write('post completed');
+        return res.end();
+    });
+}
+
+function checkForHeaders(results, expected) {
+    assert(Object.keys(expected).every(key => results[key] === expected[key]));
```

Comment:
This looks the same as `assert.strictDeepEqual`

Reply:
`results` may contain more than keys that `expected`